Assets/Scripts/Weapon/Dagger.cs

```diff
@@ -25,8 +28,18 @@ private void Slash(PlayerController executor)
             point = (Vector2) _usableItem.Player.transform.position + (point * _rayCastDist);
             Collider2D target = Physics2D.OverlapCircle(point, _rayCastRadius);
             if (target == null) return;
+            if (target.GetComponent<PlayerController>() != null)
+                HandleHit(target.GetComponent<PlayerController>(), executor);
             
             _usableItem.ReduceDurability(1);
         }
+
+        private void HandleHit(PlayerController hit, PlayerController dealer)
+        {
+            // Increase score of the dealer if hit
+            _service.PlayerManager.IncreaseScore(dealer.Stat.ID, _score);
+            // Deduct health of the hit player
+            hit.Stat.DeductHealth(_damage, dealer.Stat.ID);
```

Comment:
Consider making these functions' parameter order uniformed, like both be `PlayerID, float` so easier to read.